docs/freqai-feature-engineering.md

```diff
@@ -209,15 +209,67 @@ Another example, where the user wants to use live metrics from the trade databas
+$$ W_i = \exp(\frac{-i}{\alpha*n}) $$
 
-All other data (test data and unseen prediction data in dry/live/backtest) is always automatically normalized to the training feature space according to industry standards. FreqAI stores all the metadata required to ensure that test and prediction features will be properly normalized and that predictions are properly denormalized. For this reason, it is not recommended to eschew industry standards and modify FreqAI internals - however - advanced users can do so by inheriting `train()` in their custom `IFreqaiModel` and using their own normalization functions.
+where $W_i$ is the weight of data point $i$ in a total set of $n$ data points. Below is a figure showing the effect of different weight factors on the data points in a feature set.
+
+![weight-factor](assets/freqai_weight-factor.jpg)
 
-## Data dimensionality reduction with Principal Component Analysis
+# Building the data pipeline
+
+FreqAI uses the the [`DataSieve`](https://github.com/emergentmethods/datasieve) pipeline, which follows the SKlearn pipeline API, but adds, among other features, coherence between the X, y, and sample_weight vector point removals, and feature removal feature name following. 
+
+This means that users can use/customize any SKLearn modules and easily add them to their FreqAI data pipeline. By default, FreqAI builds the following pipeline:
+
+```py
+dk.feature_pipeline = Pipeline([
+    ('scaler', ds.DataSieveMinMaxScaler(feature_range=(-1, 1))),
+    ('di', ds.DissimilarityIndex(di_threshold=1)),
+    ])
+```
+
+But users will find that they can add PCA and other steps just by changing their configuration settings, for example, if you add `"principal_component_analysis": true` to the `feature_parameters` dict in the `freqai` config, then FreqAI will add the PCA step for you resulting in the following pipeline:
+
+```py
+dk.feature_pipeline = Pipeline([
+    ('scaler', ds.DataSieveMinMaxScaler(feature_range=(-1, 1))),
+    ('pca', ds.DataSievePCA()),
+    ('post-pca-scaler', ds.DataSieveMinMaxScaler(feature_range=(-1, 1)))
+    ('di', ds.DissimilarityIndex(di_threshold=1)),
+    ])
+```
+
+The same concept follows if users activate other config options like `"use_SVM_to_remove_outliers": true` or `"use_DBSCAN_to_remove_outliers": true`. FreqAI will add the appropriate steps to the pipeline for you.
+
+## Customizing the pipeline
+
+Users are encouraged to customize the data pipeline to their needs by building their own data pipeline. This can be done by overriding `define_data_pipeline` in their `IFreqaiModel`. For example:
+
+```py
+    def define_data_pipeline(self, dk: FreqaiDataKitchen) -> None:
+        """
+        User defines their custom eature pipeline here (if they wish)
+        """
+        from freqtrade.freqai.transforms import FreqaiQuantileTransformer
+        dk.feature_pipeline = Pipeline([
+            ('qt', FreqaiQuantileTransformer(output_distribution='normal'))
+        ])
+
+        return
```

Comment:
from an interface perspective, this (and the corresponding label_pipeline method) feels odd, and doesn't really fit into most other interface methods we have.

It's based on "create side-effects" only - while returning nothing, which is often problematic (the side-effects, not so much the "return nothing).

I feel that from a user perspective (and alignment with most other strategy/freqai interface methods), it would be better to return an array of pipeline steps  - which the backend then assembles into an actual Pipeline.

That'll also somewhat abstract the `Pipeline()` object, as the user doesn't need to know about it, really.
What the user cares about is the list of pipeline steps - not "needs to perform A, B, C".

Reply:
I agree on one point and disagree on another. 

I agree it feels better to return the pipeline, and that is what I have done. Now any user who is starting from one of our `IFreqaiModel` classes will see

```
        dk.feature_pipeline = self.define_data_pipeline()
        dk.label_pipeline = self.define_label_pipeline()
```

Which they can see is automating the construction of the `Pipeline` object. If they want, it is easy to see that they can remove the `self.define_data_pipeline()` and define it themselves following the `Pipeline` API (which is identical to SKLearn so it is a "universal" pipeline definition approach).